crates/transport_tx5/src/test.rs

```diff
@@ -0,0 +1,407 @@
+            + 'static
+            + Send
+            + Sync,
+    >,
+}
+
+impl std::fmt::Debug for CbHandler {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("CbHandler").finish()
+    }
+}
+
+impl Default for CbHandler {
+    fn default() -> Self {
+        Self {
+            new_addr: Arc::new(|_| {}),
+            peer_con: Arc::new(|_| Ok(())),
+            peer_dis: Arc::new(|_, _| {}),
+            pre_out: Arc::new(|_| Ok(bytes::Bytes::new())),
+            pre_in: Arc::new(|_, _| Ok(())),
+            space: Arc::new(|_, _, _| Ok(())),
+            module: Arc::new(|_, _, _, _| Ok(())),
+        }
+    }
+}
+
+impl TxBaseHandler for CbHandler {
+    fn new_listening_address(&self, this_url: Url) -> BoxFut<'static, ()> {
+        (self.new_addr)(this_url);
+        Box::pin(async {})
+    }
+
+    fn peer_connect(&self, peer: Url) -> K2Result<()> {
+        (self.peer_con)(peer)
+    }
+
+    fn peer_disconnect(&self, peer: Url, reason: Option<String>) {
+        (self.peer_dis)(peer, reason)
+    }
+}
+
+impl TxHandler for CbHandler {
+    fn preflight_gather_outgoing(
+        &self,
+        peer_url: Url,
+    ) -> K2Result<bytes::Bytes> {
+        (self.pre_out)(peer_url)
+    }
+
+    fn preflight_validate_incoming(
+        &self,
+        peer_url: Url,
+        data: bytes::Bytes,
+    ) -> K2Result<()> {
+        (self.pre_in)(peer_url, data)
+    }
+}
+
+impl TxSpaceHandler for CbHandler {
+    fn recv_space_notify(
+        &self,
+        peer: Url,
+        space: SpaceId,
+        data: bytes::Bytes,
+    ) -> K2Result<()> {
+        (self.space)(peer, space, data)
+    }
+}
+
+impl TxModuleHandler for CbHandler {
+    fn recv_module_msg(
+        &self,
+        peer: Url,
+        space: SpaceId,
+        module: String,
+        data: bytes::Bytes,
+    ) -> K2Result<()> {
+        (self.module)(peer, space, module, data)
+    }
+}
+
+const S1: SpaceId = SpaceId(id::Id(bytes::Bytes::from_static(b"space1")));
+
+#[tokio::test(flavor = "multi_thread")]
+async fn restart_addr() {
+    let mut test = Test::new().await;
+
+    let addr = Arc::new(Mutex::new(Vec::new()));
+    let addr2 = addr.clone();
+
+    let h = Arc::new(CbHandler {
+        new_addr: Arc::new(move |url| {
+            addr2.lock().unwrap().push(url);
+        }),
+        ..Default::default()
+    });
+    let _t = test.build_transport(h).await;
+
+    assert_eq!(1, addr.lock().unwrap().len());
+
+    test.restart().await;
+
+    tokio::time::timeout(std::time::Duration::from_secs(5), async {
+        loop {
+            if addr.lock().unwrap().len() == 2 {
+                // End the test, we're happy!
+                return;
+            }
+
+            tokio::time::sleep(std::time::Duration::from_millis(10)).await;
+        }
+    })
+    .await
+    .unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[ignore = "disconnects currently broken in tx5"]
```

Comment:
Something that needs a follow up issue?

Reply:
[already made : )](https://github.com/holochain/tx5/issues/119)